### PR TITLE
fix red herring in the readinessProbe

### PR DESCRIPTION
**Pull request by nammn**
### All Submissions:

* [x] Have you opened an Issue before filing this PR?
* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### Summary
- fixes the red herring that the agent-health-status file is missing
- continue of fix: https://github.com/mongodb/mongodb-kubernetes-operator/pull/1224, we missed a few places. This PR instead suggests to test for the file as early as possible in the code
- it can be missing, since the agent requires some time until it populates that file
- while it is still populating the probe should just return unready and it will eventually work

### CODE REVIEW

File: cmd/readiness/main.go [hunk 1]
```diff
@@ -221,7 +221,16 @@ func main() {
...
 	}
 
-	cfg, err := config.BuildFromEnvVariables(clientSet, isHeadlessMode())
+	healthStatusFilePath := config.GetEnvOrDefault(config.AgentHealthStatusFilePathEnv, config.DefaultAgentHealthStatusFilePath)
+	file, err := os.Open(healthStatusFilePath)
+	// The agent might be slow in creating the health status file.
+	// In that case, we don't want to panic to show the message
+	// in the kubernetes description. That would be a red herring, since that will solve itself with enough time.
+	if err != nil {
+		os.Exit(1)
```
Comment (lsierant) [inline cmd/readiness/main.go:230]: can we log something here to indicate what happened?

Comment (nammn) [reply cmd/readiness/main.go:230]: I was thinking about the same, but then I would need to move the logger initialization out to fit the logger: https://github.com/mongodb/mongodb-kubernetes-operator/blob/a7812ff0728930f2fb35ae4a9776b80a86e2e64b/pkg/readiness/config/config.go#L59-L64

I **could** do this 🤔

Comment (slaskawi) [reply cmd/readiness/main.go:230]: How about panicing with an error? This way it should be visible in the `kubectl get events` and `kubectl describe pod`.

Comment (slaskawi) [reply cmd/readiness/main.go:230]: Thanks for the explanation @nammn ! I believe it's still worth logging an info why this happened. Without it, investigating corner cases might be simply impossible.

Comment (nammn) [reply cmd/readiness/main.go:233]: Done

**Review by slaskawi — APPROVED** [review]

**Pull request merged**